include/jwt-cpp/jwt.h

```diff
@@ -916,6 +921,106 @@ namespace jwt {
 			const size_t signature_length;
 		};
 
+#ifndef OPENSSL110
+		/**
+		 * \brief Base class for EdDSA family of algorithms
+		 */
+		struct eddsa {
+			/**
+			 * Construct new eddsa algorithm
+			 * \param public_key EdDSA public key in PEM format
+			 * \param private_key EdDSA private key or empty string if not available. If empty, signing will always fail.
+			 * \param public_key_password Password to decrypt public key pem.
+			 * \param private_key_password Password to decrypt private key pem.
+			 * \param md Pointer to hash function
+			 * \param name Name of the algorithm
+			 */
+			eddsa(const std::string& public_key, const std::string& private_key, const std::string& public_key_password, const std::string& private_key_password, const EVP_MD* (*md)(), std::string  name)
+				: md(md), alg_name(std::move(name))
+			{
+				if (!private_key.empty()) {
+					pkey = helper::load_private_key_from_string(private_key, private_key_password);
+				}
+				else if (!public_key.empty()) {
+					pkey = helper::load_public_key_from_string(public_key, public_key_password);
+				}
+				else
+					throw ecdsa_exception(error::ecdsa_error::load_key_bio_read);
```

Comment:
should we have a new `eddsa_exception`?

Reply:
EdDSA is also a kind of elliptic curve DSA. I didn't think it necessary to specialize the error messages further.

Reply:
it would only be used once :thinking: I agree